### PR TITLE
Feature: Reimplement Indexer using built in Jina docs.matches

### LINKED ISSUE #38: bug: tries to access /results but doesn't exist (error)
**Describe the bug**
- The frontend tries to access results/results.json but /results doesn't exist (isn't tracked by the repo).

**To Reproduce**
1. clone repo
2. run backend
3. run frontend
4. see error

**Expected behavior**
- Should create /results if it's missing. 

**Screenshots**

### LINKED ISSUE #43: feat: improve indexing flow
- The indexing flow is making use of code from the chat bot example.
- Change this to make use of cleaner, more efficient indexers for the application.

References
- [AnnoyIndexer](https://github.com/jina-ai/jina-hub/blob/master/indexers/vector/AnnoyIndexer/README.md)

### PR DESCRIPTION
**Pull request by georgeamccarthy**
### PR type
- 🏆 **Enhancements** 
- 🏠 **Internal** 

### Purpose
- Refactors and reimplements indexing with Jina built in: docs.match.
- Handled slightly different response format in frontend.
- Removed protein index limit.
- Removed unnessesary saving & loading of embeddings and results where not needed.

### Why? 
- closes #43 
- closes #38 (removed unnecessary results.json)

### Feedback required over
- A quick pair of :eyes: on the code

### Mentions
- @fissoreg

### Testing
- Works on my build! :)

### CODE REVIEW

File: backend/my_executors.py [hunk 1]
```diff
@@ -90,109 +92,32 @@ def preprocessing(self, sequences: Sequence[str]) -> List[str]:
...
-        return idx, dist
+        log(f'Computing metric to {len(self._docs)} proteins.')
 
-    @property
-    def save_path(self):
-        directory = "".join(embeddings_path.split("/")[0:-1])
-        os.makedirs(directory, exist_ok=True)
-
-        return embeddings_path
+        docs.match(self._docs, metric='cosine', limit=top_k)
```
Comment (fissoreg) [inline backend/my_executors.py:118]: At a certain point, the `top_k` argument should be given as input in the request. We can also parametrize it in the frontend.

File: frontend/app.py [hunk 2]
```diff
@@ -16,7 +16,7 @@ def search(query: str, endpoint: str) -> dict:
     response = requests.post(endpoint, headers=headers, data=data)
     content = response.json()
 
-    matches = content["data"]["docs"]
+    matches = content["data"]["docs"][0]["matches"]
```
Comment (fissoreg) [inline frontend/app.py:19]: Why this? Did you notice any problems with `content["data"]["docs"]` only?

**Comment by fissoreg** [comment]
I love this PR! Got rid of quite a bit of messy code with a simple standard method.

I left a couple of minor comments, but I would like to see this merged as soon as possible.

File: frontend/app.py (hunk 2 above)
Comment (georgeamccarthy) [reply frontend/app.py:19]: `docs.matches()` appears to change `docs` in place.

It returns an array with a single element which is a dict that has key _matches_ and contains the matches to the search query.

There is no longer accessible data at content["data"]["docs"]. Have a play with `content` and see what content looks like, the structure was a tad complicated but appending `[0]["matches"]` recovered what we had before.

One could also investigate what exactly `docs.matches` is doing to `docs`.

Comment (fissoreg) [reply frontend/app.py:19]: Ok great, thanks for the explanation!

**Comment by georgeamccarthy** [comment]
Great to have the messy index code simplified! Will be much easier to test different indexing methods now.

**Pull request merged**